src/Microsoft.Extensions.FileSystemGlobbing/InMemoryDirectoryInfo.cs

```diff
@@ -0,0 +1,178 @@
+{
+    /// <summary>
+    /// Avoids using disk for uses like Pattern Matching.
+    /// </summary>
+    public class InMemoryDirectoryInfo : DirectoryInfoBase
+    {
+        private static readonly char[] DirectorySeparators = new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
+        private readonly IEnumerable<string> _files;
+
+        /// <summary>
+        /// Creates a new InMemoryDirectoryInfo with the root directory and files given.
+        /// </summary>
+        /// <param name="rootDir">The root directory that this FileSystem will use.</param>
+        /// <param name="files">Collection of file names. If relative paths <paramref name="rootDir"/> will be prepended to the paths.</param>
+        public InMemoryDirectoryInfo(string rootDir, IEnumerable<string> files)
+            : this(rootDir, files, false)
+        {
+        }
+
+        private InMemoryDirectoryInfo(string rootDir, IEnumerable<string> files, bool normalized)
+        {
+            if (string.IsNullOrEmpty(rootDir))
+            {
+                throw new ArgumentNullException(nameof(rootDir));
+            }
+
+            if (files == null)
+            {
+                files = new List<string>();
+            }
+
+            Name = Path.GetFileName(rootDir);
+            if (normalized)
+            {
+                _files = files;
+                FullName = rootDir;
+            }
+            else
+            {
+                var fileList = new List<string>(files.Count());
+
+                // normalize
+                foreach (var file in files)
+                {
+                    fileList.Add(Path.GetFullPath(file.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar)));
+                }
+
+                _files = fileList;
+
+                FullName = Path.GetFullPath(rootDir.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar));
+            }
+        }
+
+        /// <summary>
+        /// The full path for the root directory.
+        /// </summary>
+        public override string FullName { get; }
+
+        /// <summary>
+        /// The directory name for the root directory.
+        /// </summary>
+        public override string Name { get; }
+
+        /// <summary>
+        /// Gets an InMemoryDirectoryInfo that represents the
+        /// parent directory of the current InMemoryDirectoryInfo.
+        /// </summary>
+        public override DirectoryInfoBase ParentDirectory
+        {
+            get
+            {
+                return new InMemoryDirectoryInfo(Path.GetDirectoryName(FullName), _files, true);
+            }
+        }
+
+        /// <summary>
+        /// Enumerates the files and directories in the root directory.
+        /// </summary>
+        /// <returns>The list of directories and files in the root directory.</returns>
+        public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos()
```

Comment:
You can use `/// <inheritdoc />` when the [xmldoc on the base class](https://github.com/aspnet/FileSystem/blob/dev/src/Microsoft.Extensions.FileSystemGlobbing/Abstractions/DirectoryInfoBase.cs#L13-L17) is sufficient to describe this overload.